src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -7487,6 +7487,9 @@ we will have to use something else.
             throw SemanticError("Only Name or Attribute type supported in Call",
                 x.base.base.loc);
         }
+        if (call_name == "type" && x.n_args == 0) {
+            throw SemanticError("type() takes 1 required argument `object`", x.base.base.loc);
+        }
```

Comment:
```suggestion
```

Reply:
This check is required as an unhandled exception is thrown for an empty function call.

Reply:
I think every semantics checks related to `type(..)` needs to be handled inside the `create_ObjectType` function.

Reply:
It's not @Thirumalai-Shaktivel. That's why I had to add this here in the call visitor :(

Reply:
Share an minimum reproducible example, I will check it out.

Reply:
Do you mean for the following example?
```py
print(type())
```

CPython and LPython both throws following errors:
```console
$ python examples/expr2.py 
Traceback (most recent call last):
  File "/Users/thirumalai/Open_Source/lpython/examples/expr2.py", line 1, in <module>
    print(type())
TypeError: type() takes 1 or 3 arguments
$ lpython examples/expr2.py 
semantic error: type() takes 1 required argument `object`
 --> examples/expr2.py:1:7
  |
1 | print(type())
  |       ^^^^^^ 

```

Reply:
```python
print(type())
```

Reply:
My bad, yea, it throws an exception:
```console
....
  File "/Users/thirumalai/Open_Source/lpython/src/libasr/asr.h", line 49205
    LCOMPILERS_ASSERT(f != nullptr);
AssertFailed: f != nullptr
```

Reply:
Apply the following diff to fix the bug:
```diff
diff --git a/src/lpython/semantics/python_ast_to_asr.cpp b/src/lpython/
semantics/python_ast_to_asr.cpp
index 714349d7c..9ae12641d 100644
--- a/src/lpython/semantics/python_ast_to_asr.cpp
+++ b/src/lpython/semantics/python_ast_to_asr.cpp
@@ -7487,9 +7487,6 @@ we will have to use something else.
             throw SemanticError("Only Name or Attribute type supported
 in Call",
                 x.base.base.loc);
         }
-        if (call_name == "type" && x.n_args == 0) {
-            throw SemanticError("type() takes 1 required argument `obj
ect`", x.base.base.loc);
-        }
 
         ASR::symbol_t *s = current_scope->resolve_symbol(call_name);
         if( s && ASR::is_a<ASR::Module_t>(*s) ) {
@@ -7522,7 +7519,7 @@ we will have to use something else.
                 }
                 Vec<ASR::expr_t*> args_; args_.reserve(al, x.n_args);
                 visit_expr_list(x.m_args, x.n_args, args_);
-                if (ASRUtils::is_array(ASRUtils::expr_type(args_[0])) 
&&
+                if (x.n_args && ASRUtils::is_array(ASRUtils::expr_type
(args_[0])) &&
                     imported_functions[call_name] == "math" ) {
                     throw SemanticError("Function '" + call_name + "' 
does not accept vector values",
                         x.base.base.loc);
```